kythe/cxx/verifier/souffle_interpreter.cc

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2020 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kythe/cxx/verifier/souffle_interpreter.h"
+
+#include <array>
+#include <memory>
+
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_cat.h"
+#include "glog/logging.h"
+#include "interpreter/Engine.h"
+#include "souffle/RamTypes.h"
+#include "souffle/io/IOSystem.h"
+#include "third_party/souffle/parse_transform.h"
+
+namespace kythe::verifier {
+namespace {
+constexpr std::array<int, 4> kInputData = {1, 2, 2, 3};
+}  // anonymous namespace
```

Comment:
everything not mentioned in the header should be enclosed by the anonymous namespace: https://abseil.io/tips/186